modules/ducktests/tests/ignitetest/services/utils/control_utility.py

```diff
@@ -0,0 +1,167 @@
+
+from ducktape.cluster.remoteaccount import RemoteCommandError
+
+
+class ControlUtility:
+    """
+    Control utility (control.sh) wrapper.
+    """
+    BASE_COMMAND = "control.sh"
+
+    def __init__(self, cluster, text_context):
+        self._cluster = cluster
+        self.logger = text_context.logger
+
+    def baseline(self):
+        """
+        :return Baseline nodes.
+        """
+        return self.cluster_state().baseline
+
+    def cluster_state(self):
+        """
+        :return: Cluster state.
+        """
+        output = self.__run("--baseline")
+
+        return self.__parse_cluster_state(output)
+
+    def set_baseline(self, baseline):
+        """
+        :param baseline: Baseline nodes or topology version to set as baseline.
+        """
+        if isinstance(baseline, int):
+            result = self.__run("--baseline version %d --yes" % baseline)
+        else:
+            result = self.__run("--baseline set %s --yes" %
+                                ",".join([node.account.externally_routable_ip for node in baseline]))
+
+        return self.__parse_cluster_state(result)
+
+    def add_to_baseline(self, nodes):
+        """
+        :param nodes: Nodes that should be added to baseline.
+        """
+        result = self.__run("--baseline add %s --yes" %
+                            ",".join([node.account.externally_routable_ip for node in nodes]))
+
+        return self.__parse_cluster_state(result)
+
+    def remove_from_baseline(self, nodes):
+        """
+        :param nodes: Nodes that should be removed to baseline.
+        """
+        result = self.__run("--baseline remove %s --yes" %
+                            ",".join([node.account.externally_routable_ip for node in nodes]))
+
+        return self.__parse_cluster_state(result)
+
+    def disable_baseline_auto_adjust(self):
+        """
+        Disable baseline auto adjust.
+        """
+        return self.__run("--baseline auto_adjust disable --yes")
+
+    def enable_baseline_auto_adjust(self, timeout=None):
+        """
+        Enable baseline auto adjust.
+        :param timeout: Auto adjust timeout in millis.
+        """
+        timeout_str = "timeout %d" % timeout if timeout else ""
+        return self.__run("--baseline auto_adjust enable %s --yes" % timeout_str)
+
+    def activate(self):
+        """
+        Activate cluster.
+        """
+        return self.__run("--activate --yes")
+
+    def deactivate(self):
+        """
+        Deactivate cluster.
+        """
+        return self.__run("--deactivate --yes")
+
+    @staticmethod
+    def __parse_cluster_state(output):
+        state_pattern = re.compile("Cluster state: ([^\\s]+)")
+        topology_pattern = re.compile("Current topology version: (\\d+)")
+        baseline_pattern = re.compile("Consistent(Id|ID)=([^\\s]+),\\sS(tate|TATE)=([^\\s]+),?(\\sOrder=(\\d+))?")
+
+        match = state_pattern.search(output)
+        state = match.group(1) if match else None
```

Comment:
What do you think, is it better to use named group instead of indices? 
https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups